sensor/kubernetes/fake/deployment.go

```diff
@@ -6,18 +6,84 @@ import (
 	"math/rand"
 	"time"
 
+	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/concurrency"
 	"github.com/stackrox/rox/pkg/containerid"
 	"github.com/stackrox/rox/pkg/fixtures"
 	"github.com/stackrox/rox/pkg/k8sutil"
 	"github.com/stackrox/rox/pkg/kubernetes"
 	"github.com/stackrox/rox/pkg/pointers"
+	"github.com/stackrox/rox/pkg/sync"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+var (
+	processPool = newProcessPool()
+)
+
+// ProcessPool stores processes by containerID using a map
+type ProcessPool struct {
+	Processes map[string][]*storage.ProcessSignal
+	Capacity  int
+	lock      sync.RWMutex
+}
+
+func newProcessPool() *ProcessPool {
+	return &ProcessPool{
+		Processes: make(map[string][]*storage.ProcessSignal),
+		Capacity:  10000,
+	}
+}
+
+func (p *ProcessPool) getProcessPoolSize() int {
+	size := 0
```

Comment:
is it possible to just keep track of this in a variable instead of computing it?

Reply:
Done